packages/talisman-ui/src/components/MysticalBackgroundV3/useCelestialArtifact.ts

```diff
@@ -1,95 +1,175 @@
   )
 
-  const [characteristics, setCharacteristics] = useState<ArtifactCharacteristics>(
-    generateCharacteristics(config, parentSize, duration, refInitialized.current, color)
+  const refInitialized = useRef(false)
+  const generate = useCallback(
+    () =>
+      generateArtifact({
+        config,
+        parentSize,
+        artifactIndex,
+        color,
+        duration,
+        initialized: refInitialized.current,
+      }),
+    [color, config, duration, artifactIndex, parentSize]
   )
 
-  useEffect(() => {
+  const [artifact, setArtifact] = useState(generate)
+  useLayoutEffect(() => {
+    let active = true
     const update = () => {
-      setCharacteristics(
-        generateCharacteristics(config, parentSize, duration, refInitialized.current, color)
-      )
+      if (!active) return
+
+      setArtifact(generate())
       refInitialized.current = true
-    }
 
-    const interval = setInterval(update, duration)
+      setTimeout(update, duration)
+    }
 
-    // change after 10ms to ensure the first render has occured
-    setTimeout(() => update(), 10)
+    // On mount, useState will have an initial position & size & color for the artifact.
+    //
+    // Then once that initial state has rendered (thanks to useLayoutEffect) we will begin our first
+    // transition to the second position & size & color by calling `update` here.
+    //
+    // For this first transition, `initialized` will be `false` - so we will get a slightly different
+    // transition timing function (ease-in-out instead of ease-out).
+    //
+    // After that, `update` will call itself every `duration` interval, up until we unmount and set `active` to false.
+    update()
 
     return () => {
-      clearInterval(interval)
+      active = false
     }
-  }, [config, duration, parentSize, color])
+  }, [generate, duration])
 
-  return characteristics
+  return artifact
+}
+
+export type ParentSize = {
+  width: number
+  height: number
 }
+
+type GenerateArtifactProps = CelestialArtifactProps & {
+  /** The duration between each artifact position, scale and color change */
+  duration: number
+
+  /** Whether or not the artifact has been initialized */
+  initialized: boolean
+}
+
+const generateArtifact = ({
+  config,
+  parentSize,
+  artifactIndex,
+  color: forceColor,
+  duration,
+  initialized,
+}: GenerateArtifactProps): CSSProperties => {
+  const maxSize = Math.min(parentSize.width, parentSize.height) * 2
+  const width = Math.round(maxSize * randBetween(config.radiusMin, config.radiusMax))
+  const height = Math.round(width * (1 - config.ellipsisRatio * Math.random()))
+
+  const x = Math.round(Math.random() * parentSize.width)
+  const y = Math.round(Math.random() * parentSize.height - Math.max(width / 2, height / 2) / 2)
+
+  const color = forceColor ?? Color.hsv(Math.random() * 360, 100, 100).hex()
+  const opacity = randBetween(config.opacityMin, config.opacityMax)
+  const background = artifactBg(color, opacity)
```

Comment:
We can't set the `opacity` on the artifact itself, because that's the prop we're using to transition between the two artifact colors!

So - I've just baked the intended opacity into the `background` gradient.